app/src/main/java/org/wikipedia/usercontrib/UserContribStats.kt

```diff
@@ -19,28 +21,26 @@ object UserContribStats {
 
     private var totalEdits: Int = 0
     var totalReverts: Int = 0
-    var totalDescriptionEdits: Int = 0
-    var totalImageCaptionEdits: Int = 0
-    var totalImageTagEdits: Int = 0
-
-    suspend fun verifyEditCountsAndPauseState() {
-        val response = ServiceFactory.get(Constants.wikidataWikiSite).getEditorTaskCounts()
-        if (response.query?.userInfo?.isBlocked != true) {
-            response.query?.editorTaskCounts?.let {
-                totalEdits = it.totalEdits
-                totalDescriptionEdits = it.totalDescriptionEdits
-                totalImageCaptionEdits = it.totalImageCaptionEdits
-                totalImageTagEdits = it.totalDepictsEdits
-                totalReverts = it.totalReverts
-                maybePauseAndGetEndDate()
-            }
-        }
+
+    fun verifyEditCountsAndPauseState(totalContributionsList: List<UserContribution>) {
+        totalEdits = totalContributionsList.size
+        totalReverts = totalContributionsList.count { it.ns == 0 && it.tags.contains("mw-reverted") }
+        maybePauseAndGetEndDate()
     }
 
-    suspend fun getPageViews(response: MwQueryResponse): Long {
+    suspend fun getPageViews(homeWikiContributions: List<UserContribution>, wikidataContributions: List<UserContribution>): Long {
```

Comment:
I noticed you removed the edit counts for image captions and image tags and did not include the contributions in this pageView function. Is this expected?

Reply:
Yes. Those pageviews are definitely meaningless. (It would get the pageviews of _the Commons page_ of each image, which are not normally visited by anyone.)